pkg/ipcache/ipcache.go

```diff
@@ -179,7 +179,10 @@ func NewIPCache(c *Configuration) *IPCache {
 
 // Shutdown cleans up asynchronous routines associated with the IPCache.
 func (ipc *IPCache) Shutdown() error {
-	return ipc.controllers.RemoveControllerAndWait(LabelInjectorName)
+	if ipc.controllers.Lookup(LabelInjectorName) != nil {
```

Comment:
Isn't this racy? If the controller is added after the lookup happens we now leak it, no? Should instead RemoveControllerAndWait be idempotent against controllers which don't exist?